### ISSUE TITLE
Default file_cache_path with local-mode may break things and do not honor documentation

### ISSUE THREAD

**Issue by gionn** [chef/chef#2875]
Hi,

I've just found a misbehaving ark resource due to the default relative path set as file_cache_path:

```
./local-mode-cache/cache
```

Also, the documentation wrongly states:

```
When running the chef-client using the --local-mode option, ~/.chef/local-mode-cache is the default value for cache_path.
```

that looks a good path, anyway.

I was running with user root on ~:

```
chef-client -z -c client.rb
```

A workaround is to configure _file_cache_path_ in _client.rb_, e.g.:

```
data_bag_path "/tmp/chef/data_bags"
environment_path "/tmp/chef/environments"
cookbook_path "/tmp/chef/cookbooks"
file_cache_path "/tmp/chef/cache"
ssl_verify_mode :verify_peer
```

**Comment by gionn** [comment]
I made a quick dig into it, and chef relies on File.dirname passing the configuration file as passed on the command-line, so I am getting a relative cache path since I passed a relative path for the configuration file.

It may be a good idea to always return an absolute path?

**Comment by lamont-granquist** [comment]
think this is a dup of another issue as well, but offhand i couldn't find it.

**Comment by vsingh-msys** [comment]
Fixed #2875 via #9042
